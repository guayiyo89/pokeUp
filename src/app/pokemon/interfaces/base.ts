export interface Base {
    name: string,
    url: string
}

export interface MainBase {
    count: number;
    next?: any;
    previous?: any;
    results: Base[];
}
