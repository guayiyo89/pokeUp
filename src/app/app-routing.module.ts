import { NgModule } from '@angular/core';
import { RouterModule, Routes, PreloadAllModules } from '@angular/router';
import { ErrorScreenComponent } from './shared/components/error-screen/error-screen.component';

const routes: Routes = [
  {
    path: 'buscar-pokemon',
    loadChildren: () =>
      import('./pokemon/pokemon.module').then((module) => module.PokemonModule),
  }
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, {
      preloadingStrategy: PreloadAllModules,
      relativeLinkResolution: 'legacy',
    }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
