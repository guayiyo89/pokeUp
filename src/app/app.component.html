lalalal
<router-outlet></router-outlet>