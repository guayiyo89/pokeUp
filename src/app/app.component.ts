import { Component, OnInit } from '@angular/core';
import { LoadingService } from './core/services/loading.service';
import { PokemonService } from './pokemon/services/pokemon.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  title = 'pokedex-angular';

  constructor(private pokeSvc: PokemonService, public loadingSvc: LoadingService) {}

  ngOnInit() {
    this.loadPokemon();
    this.loadHabitat();
    this.loadRegions();
    this.loadTypes();
  }

  loadPokemon() {
    this.loadingSvc.showLoadingScreen(true)
    this.pokeSvc
      .getPokemons1stGen()
      .subscribe((resp) => this.pokeSvc.saveData(resp.pokemon_species));
  }

  loadHabitat() {
    this.loadingSvc.showLoadingScreen(true)
    this.pokeSvc
      .getHabitat()
      .subscribe((resp) => this.pokeSvc.saveHabitatData(resp.results));
  }

  loadTypes() {
    this.pokeSvc
      .getTypes()
      .subscribe((resp) => this.pokeSvc.saveTypeData(resp.results));
  }

  loadRegions() {
    this.pokeSvc
      .getRegions()
      .subscribe((resp) => this.pokeSvc.saveRegionData(resp.results));
  }
}
