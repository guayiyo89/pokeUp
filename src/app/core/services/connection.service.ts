import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class ConnectionService {

  constructor() { }

  offline$ = new Subject<boolean>();

  showConnectionError() {
    this.offline$.next(true);
  }

  hideConnectionError() {
    this.offline$.next(false);
  }
}
