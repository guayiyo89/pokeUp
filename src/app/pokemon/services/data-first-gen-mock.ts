import { MainGeneration } from "../interfaces/generation-i.interface";

export const data1stGenMock: MainGeneration = {
    "abilities": [],
    "id": 1,
    "main_region": {
        "name": "kanto",
        "url": "https://pokeapi.co/api/v2/region/1/"
    },
    "moves": [
        {
            "name": "pound",
            "url": "https://pokeapi.co/api/v2/move/1/"
        },
        {
            "name": "karate-chop",
            "url": "https://pokeapi.co/api/v2/move/2/"
        },
        {
            "name": "double-slap",
            "url": "https://pokeapi.co/api/v2/move/3/"
        }
    ],
    "name": "generation-i",
    "names": [
        {
            "language": {
                "name": "ja-Hrkt",
                "url": "https://pokeapi.co/api/v2/language/1/"
            },
            "name": "だいいちせだい"
        },
    ],
    "pokemon_species": [
        {
            "name": "bulbasaur",
            "url": "https://pokeapi.co/api/v2/pokemon-species/1/"
        },
        {
            "name": "staryu",
            "url": "https://pokeapi.co/api/v2/pokemon-species/120/"
        },
        {
            "name": "scyther",
            "url": "https://pokeapi.co/api/v2/pokemon-species/123/"
        },
        {
            "name": "pinsir",
            "url": "https://pokeapi.co/api/v2/pokemon-species/127/"
        },
        {
            "name": "slowbro",
            "url": "https://pokeapi.co/api/v2/pokemon-species/80/"
        },
        {
            "name": "magneton",
            "url": "https://pokeapi.co/api/v2/pokemon-species/82/"
        },
        {
            "name": "dodrio",
            "url": "https://pokeapi.co/api/v2/pokemon-species/85/"
        },
        {
            "name": "dewgong",
            "url": "https://pokeapi.co/api/v2/pokemon-species/87/"
        },
        {
            "name": "muk",
            "url": "https://pokeapi.co/api/v2/pokemon-species/89/"
        }
    ],
    "types": [
        {
            "name": "normal",
            "url": "https://pokeapi.co/api/v2/type/1/"
        },
        {
            "name": "fighting",
            "url": "https://pokeapi.co/api/v2/type/2/"
        },
        {
            "name": "rock",
            "url": "https://pokeapi.co/api/v2/type/6/"
        },
        {
            "name": "bug",
            "url": "https://pokeapi.co/api/v2/type/7/"
        },
        {
            "name": "ghost",
            "url": "https://pokeapi.co/api/v2/type/8/"
        },
        {
            "name": "fire",
            "url": "https://pokeapi.co/api/v2/type/10/"
        },
        {
            "name": "water",
            "url": "https://pokeapi.co/api/v2/type/11/"
        }
    ],
    "version_groups": [
        {
            "name": "red-blue",
            "url": "https://pokeapi.co/api/v2/version-group/1/"
        },
        {
            "name": "yellow",
            "url": "https://pokeapi.co/api/v2/version-group/2/"
        }
    ]
}