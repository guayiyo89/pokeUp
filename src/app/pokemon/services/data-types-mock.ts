import { MainBase } from "../interfaces/base"

export const mock: MainBase = {
    count: 2,
    next: null,
    previous: null,
    results: [
        {
            "name": "cave",
            "url": "https://pokeapi.co/api/v2/pokemon-habitat/1/"
        },
        {
            "name": "forest",
            "url": "https://pokeapi.co/api/v2/pokemon-habitat/2/"
        }
    ]
}