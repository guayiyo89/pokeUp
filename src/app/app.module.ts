import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { PokemonModule } from './pokemon/pokemon.module';
import { HeaderComponent } from './shared/components/header/header.component';
import { FooterComponent } from './shared/components/footer/footer.component';
import { NotFoundComponent } from './shared/components/not-found/not-found.component';
import { LoadingScreenComponent } from './shared/components/loading-screen/loading-screen.component';
import { ErrorScreenComponent } from './shared/components/error-screen/error-screen.component';
import { ErrorHandlerInterceptor } from './core/interceptors/error-handler.interceptor';
import { ConnectionComponent } from './core/components/connection/connection.component';
import { ConnectionInterceptor } from './core/interceptors/connection.interceptor';

@NgModule({
  declarations: [
    AppComponent,
    HeaderComponent,
    FooterComponent,
    NotFoundComponent,
    LoadingScreenComponent,
    ErrorScreenComponent,
    ConnectionComponent
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    PokemonModule,
    AppRoutingModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: ErrorHandlerInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: ConnectionInterceptor, multi: true }
  ],
  bootstrap: [AppComponent],
  schemas: [CUSTOM_ELEMENTS_SCHEMA]
})
export class AppModule { }
