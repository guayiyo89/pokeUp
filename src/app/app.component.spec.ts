import { HttpClientTestingModule } from '@angular/common/http/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';
import { AppComponent } from './app.component';
import { MainBase } from './pokemon/interfaces/base';
import { MainGeneration } from './pokemon/interfaces/generation-i.interface';
import { data1stGenMock } from './pokemon/services/data-first-gen-mock';
import { mock } from './pokemon/services/data-types-mock';
import { PokemonService } from './pokemon/services/pokemon.service';

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let pokeSvc: PokemonService;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule, HttpClientTestingModule],
      declarations: [AppComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    compiled = fixture.nativeElement;
    pokeSvc = TestBed.inject(PokemonService);
    fixture.detectChanges()
  });

  describe('When initializing', () => {
    test('#should create the component', () => {
      expect(component).toBeTruthy();
    });

    test('#should call the loadPokemon function', () => {
      const spy = jest.spyOn(component, 'loadPokemon');
      component.ngOnInit();
      expect(spy).toHaveBeenCalled();
    });

    test('#should call the loadRegions function', () => {
      const spy = jest.spyOn(component, 'loadRegions');
      component.ngOnInit();
      expect(spy).toHaveBeenCalled();
    });

    test('#should call the loadTypes function', () => {
      const spy = jest.spyOn(component, 'loadTypes');
      component.ngOnInit();
      expect(spy).toHaveBeenCalled();
    });

    test('#should call the loadHabitat function', () => {
      const spy = jest.spyOn(component, 'loadHabitat');
      component.ngOnInit();
      expect(spy).toHaveBeenCalled();
    });

    test('#should get the data from loadPokemon and then save it', () => {
      const data: MainGeneration = data1stGenMock;
      const service = TestBed.inject(PokemonService);
      const spy = jest.spyOn(service, 'saveData');
      jest
        .spyOn(service, 'getPokemons1stGen')
        .mockReturnValue(of(data as MainGeneration));

      component.ngOnInit();
      component.loadPokemon();

      service
        .getPokemons1stGen()
        .subscribe((value) => {
          expect(value).toBe(data)
          expect(spy).toHaveBeenCalled
          expect(service.dataPokemon).toBe(data.pokemon_species)
        });
    });

    test('#should get the data from loadHabitat and then save it', () => {
      const data: MainBase = mock;
      const service = TestBed.inject(PokemonService);
      const spy = jest.spyOn(service, 'saveHabitatData');
      jest
        .spyOn(service, 'getHabitat')
        .mockReturnValue(of(data as MainBase));

      component.ngOnInit();
      component.loadHabitat()

      service
        .getHabitat()
        .subscribe((value) => {
          expect(value).toBe(data)
          expect(spy).toHaveBeenCalled
          expect(service.dataHabitat).toBe(data.results)
        });
    });

    test('#should get the data from loadTypes and then save it', () => {
      const data: MainBase = mock;
      const service = TestBed.inject(PokemonService);
      const spy = jest.spyOn(service, 'saveTypeData');
      jest
        .spyOn(service, 'getTypes')
        .mockReturnValue(of(data as MainBase));

      component.ngOnInit();
      component.loadTypes()

      service
        .getTypes()
        .subscribe((value) => {
          expect(value).toBe(data)
          expect(spy).toHaveBeenCalled
          expect(service.dataTypes).toBe(data.results)
        });
    });

    test('#should get the data from loadRegions and then save it', () => {
      const data: MainBase = mock;
      const service = TestBed.inject(PokemonService);
      const spy = jest.spyOn(service, 'saveRegionData');
      jest
        .spyOn(service, 'getRegions')
        .mockReturnValue(of(data as MainBase));

      component.ngOnInit();
      component.loadRegions()

      service
        .getRegions()
        .subscribe((value) => {
          expect(value).toBe(data)
          expect(spy).toHaveBeenCalled
          expect(service.dataRegion).toBe(data.results)
        });
    });

    test('#should do match with snapshot', () => {
      expect(compiled.innerHTML).toMatchSnapshot();
    });
  });
});
