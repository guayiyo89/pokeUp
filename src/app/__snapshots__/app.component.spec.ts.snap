// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent When initializing #should do match with snapshot 1`] = `
"<app-header></app-header><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><div><router-outlet></router-outlet></div><app-connection></app-connection><app-footer></app-footer>"
`;
